fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/mapping/RelationMapping.java

```diff
@@ -0,0 +1,126 @@
+        }
+        LinkedListMultimap<String, MappedRelation> targetTableRelationIdMap = LinkedListMultimap.create();
+        for (CatalogRelation relation : targets) {
+            String tableQualifier = getTableQualifier(relation.getTable());
+            if (tableQualifier == null) {
+                return null;
+            }
+            targetTableRelationIdMap.put(tableQualifier, MappedRelation.of(relation.getRelationId(), relation));
+        }
+
+        Set<String> sourceTableKeySet = sourceTableRelationIdMap.keySet();
+        List<List<Pair<MappedRelation, MappedRelation>>> mappedRelations = new ArrayList<>();
+
+        for (String sourceTableQualifier : sourceTableKeySet) {
+            List<MappedRelation> sourceMappedRelations = sourceTableRelationIdMap.get(sourceTableQualifier);
+            List<MappedRelation> targetMappedRelations = targetTableRelationIdMap.get(sourceTableQualifier);
+            if (targetMappedRelations.isEmpty()) {
+                continue;
+            }
+            // if source and target relation appear once, just map them
+            if (targetMappedRelations.size() == 1 && sourceMappedRelations.size() == 1) {
+                mappedRelations.add(ImmutableList.of(Pair.of(sourceMappedRelations.get(0),
+                        targetMappedRelations.get(0))));
+                continue;
+            }
+            // relation appear more than once, should cartesian them
+            ImmutableList<Pair<MappedRelation, MappedRelation>> relationMapping = Lists.cartesianProduct(
+                            sourceTableRelationIdMap.get(sourceTableQualifier), targetMappedRelations)
+                    .stream()
+                    .map(listPair -> Pair.of(listPair.get(0), listPair.get(1)))
+                    .collect(ImmutableList.toImmutableList());
+            mappedRelations.add(relationMapping);
+        }
+
+        int mappedRelationCount = mappedRelations.size();
+
+        return Lists.cartesianProduct(mappedRelations).stream()
+                .map(mappedRelationList -> {
+                    BiMap<MappedRelation, MappedRelation> relationMappedRelationBiMap =
+                            HashBiMap.create(mappedRelationCount);
+                    for (int relationIndex = 0; relationIndex < mappedRelationCount; relationIndex++) {
+                        relationMappedRelationBiMap.put(mappedRelationList.get(relationIndex).key(),
+                                mappedRelationList.get(relationIndex).value());
+                    }
+                    return RelationMapping.of(relationMappedRelationBiMap);
+                })
+                .collect(ImmutableList.toImmutableList());
+    }
+
+    private static String getTableQualifier(TableIf tableIf) {
+        String tableName = tableIf.getName();
+        DatabaseIf database = tableIf.getDatabase();
+        if (database == null) {
+            return null;
+        }
+        return database.getFullName() + ":" + tableName;
```

Comment:
Use tableId may be better. It's globally unique

Reply:
OK, you are right, have fixed it